pkgs/servers/softether/4.38.nix

```diff
@@ -0,0 +1,38 @@
+{ lib, stdenv, fetchurl
+, openssl, readline, ncurses, zlib
+, dataDir ? "/var/lib/softether" }:
+
+stdenv.mkDerivation rec {
+  pname = "softether";
+  version = "4.38";
+  build = "9760";
+
+  src = fetchurl {
+    url = "https://github.com/SoftEtherVPN/SoftEtherVPN_Stable/releases/download/v${version}-${build}-rtm/softether-src-v${version}-${build}-rtm.tar.gz";
+    sha256 = "0d8zahi9lkv72jh8yj66pwrsi4451vk113d3khzrzgbic6s2i0g6";
+  };
+
+  buildInputs = [ openssl readline ncurses zlib ];
+
+  preConfigure = ''
+      ./configure
+  '';
+
+  buildPhase = ''
+      mkdir -p $out/bin
+      sed -i \
+          -e "/INSTALL_BINDIR=/s|/usr/bin|/bin|g" \
+          -e "/_DIR=/s|/usr|${dataDir}|g" \
+          -e "s|\$(INSTALL|$out/\$(INSTALL|g" \
+          -e "/echo/s|echo $out/|echo |g" \
+          Makefile
```

Comment:
```suggestion
    ./configure
  '';

  buildPhase = ''
    mkdir -p $out/bin
    sed -i \
      -e "/INSTALL_BINDIR=/s|/usr/bin|/bin|g" \
      -e "/_DIR=/s|/usr|${dataDir}|g" \
      -e "s|\$(INSTALL|$out/\$(INSTALL|g" \
      -e "/echo/s|echo $out/|echo |g" \
      Makefile
```

Reply:
I can't sure v4.38 is working incorrectly, so keep the old version.

Reply:
@SuperSandro2000 in [Nixpkgs Users and Contributors Guide -- 14.1 Syntax](https://jtojnar.github.io/dumpling/#sec-syntax)

> Use 2 spaces of indentation per indentation level in Nix expressions, 4 spaces in shell scripts.

do we need to follow this rule?

Reply:
> do we need to follow this rule?

Yes. This is a nix file.

> I can't sure v4.38 is working incorrectly, so keep the old version.

Most people are probably using `softether`. Please test it and if it works, then it works. We don't need to keep two versions just in case  that something might be. You can always rollback or we fix issues in a few days.

Reply:
> Yes. This is a nix file.
But `buildPhase` contacts shell script, right?

Reply:
Yes, technically but it is still intended with 2 spaces. 4 spaces is for .sh files.

Reply:
Ok, I get it. Thanks for your help.